c10/core/impl/DeviceGuardImplInterface.h

```diff
@@ -139,6 +139,14 @@ struct C10_API DeviceGuardImplInterface {
    */
   virtual Stream exchangeStream(Stream) const noexcept = 0;
 
+  /**
+   * Set the current stream to Stream, You are NOT required to set the current
+   * device to match the device of this stream.
+   */
+  virtual void setStream(Stream) const {
```

Comment:
`exchangeStream` is sufficient, but an extra instruction to obtain the current stream first compared to `setStream`.

Reply:
Do you have use case where this overhead is measurably too much for your use case? If so, then we can add it. If not, let's delay adding this until we have a use case for it to avoid premature optimization.

Reply:
OK, we will reuse `exchangeStream` first to avoid adding a new one.